plugin/caffe/caffe_data_iter.cc

```diff
@@ -0,0 +1,217 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file caffe_data_iter.cc
+ * \brief register mnist iterator
+*/
+#include <caffe/proto/caffe.pb.h>
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator.h>
+#include <atomic>
+
+#include "../../src/operator/operator_common.h"
+#include "caffe_common.h"
+#include "caffe_stream.h"
+#include "caffe_fieldentry.h"
+#include "caffe_blob.h"
+#include "../../src/io/inst_vector.h"
+#include "../../src/io/iter_prefetcher.h"
+#include "../../src/operator/cast-inl.h"
+
+namespace mxnet {
+namespace io {
+
+struct CaffeDataParam : public dmlc::Parameter<CaffeDataParam> {
+  /*! \brief protobuf text */
+  ::caffe::LayerParameter prototxt;
+  /*! \brief maximum iteration count */
+  int max_iterations;
```

Comment:
Actually, elsewhere epoch_size seems to be # batches to run rather than number of items.  args.num_examples seems to be the correct variable.  I will change.